src/pcre2_substitute.c

```diff
@@ -1194,8 +1192,17 @@ if (!replacement_only)
   CHECKMEMCPY(subject + start_offset, fraglength);
   }
 
-temp[0] = 0;
-CHECKMEMCPY(temp, 1);
+/* Zero terminate the output string, truncating if needed. */
+
+if (lengthleft-- > 0) ++buff_offset;
+else
+  {
+  if ((suboptions & PCRE2_SUBSTITUTE_OVERFLOW_LENGTH) == 0) goto NOROOM;
+  overflowed = TRUE;
+  extra_needed++;
+  lengthleft = 0;
+  }
+if (!overflowed || lengthleft == 0) buffer[buff_offset] = 0; else extra_needed++;
```

Comment:
Could you explain why you need to inline the CHECKMEMCPY here for, for the trailing NUL?

What was wrong before? Do you want the returned string to be NUL-terminated, even if the function returns an error?

Reply:
> Do you want the returned string to be NUL-terminated, even if the function returns an error?

Correct, I found the way this function behaves strange and the fact that it will return non NUL terminated strings on overflow, potentially risky.